fire-hose-management-frontend/src/components/Departments/EditDepartmentOverlay.jsx

```diff
@@ -0,0 +1,186 @@
+import {CTextField} from "../shared/input/CTextField";
+
+const useStyles = makeStyles((theme) => ({
+    checkbox: {
+        display: "table-cell",
+        verticalAlign: "middle"
+    },
+    footer: {
+        paddingTop: "24px",
+        paddingBottom: "32px"
+    }
+}));
+
+const genderMenuItems = [<MenuItem key={0} value={"MALE"}>Herr</MenuItem>,
+    <MenuItem key={1} value={"FEMALE"}>Frau</MenuItem>,
+    <MenuItem key={2} value={"OTHER"}>Divers</MenuItem>]
+
+const EditDepartmentOverlay = ({
+                                   department,
+                                   handleClose
+                               }) => {
+
+    const classes = useStyles();
+
+    const [gender, setGender] = useState(department.contact.gender)
+    const [firstname, setFirstname] = useState(department.contact.firstname)
+    const [lastname, setLastname] = useState(department.contact.lastname)
+    const [mail, setMail] = useState(department.contact.mail)
+    const [sendConfirmationMail, setSendConformationMail] = useState(department.sendConfirmationMail)
+
+    const editDepartmentMutation = useMutation((departmentDto) => putDepartment(department.id, departmentDto), {
+        onSuccess: () => {
+            Notifications.showSuccess("Die Abteilung wurde erfolgreich aktualisiert!")
+
+        },
+        onError: (error) => {
+            Notifications.showError("Die Abteilung konnte nicht aktualisiert werden!")
+            console.log(error)
+        }
+    })
+
+
+    const departmentBuilder = (gender, firstname, lastname, mail, sendConfirmationMail) => {
+        return {
+            gender: gender,
+            firstname: firstname,
+            lastname: lastname,
+            mail: mail,
+            sendConfirmationMail: sendConfirmationMail
+        }
+    }
+
+
+    console.log(department)
+
+    return (<Overlay
+        onClose={() => handleClose()}
+        headerContent={<h2>{department.departmentName} aktualisieren</h2>}
+        size={"m"}>
+        <ValidatorForm
+            onSubmit={() => editDepartmentMutation.mutate(departmentBuilder(gender, firstname, lastname, mail, sendConfirmationMail))}>
+            <Grid container spacing={3}>
+                <Grid item xs={12}>
+                    <TextField
+                        label={"Abteilung"}
+                        value={department.departmentName}
+                        variant="filled"
+                        fullWidth={true}
+                        disabled={true}
+                    />
+                </Grid>
+                <Grid item md={6}>
+                    <TextField
+                        label={"Straße"}
+                        value={department.street}
+                        variant="filled"
+                        fullWidth={true}
+                        disabled={true}
+                    />
+                </Grid>
+                <Grid item md={6}>
+                    <TextField
+                        label={"Hausnummer"}
```

Comment:
Codacy has a fix for the issue: [Replace `························` with `··············`](https://app.codacy.com/gh/StitZle/fire-hose-complete/pullRequest?prid=9542260)
```suggestion
              label={"Hausnummer"}
```